### ISSUE TITLE
linuxkit push vcenter segmentation violation

### ISSUE THREAD

**Issue by firecyberice** [linuxkit/linuxkit#2635]
`linuxkit push vcenter` exits with segmentation violation because of a nil pointer reference.

**Steps to reproduce the issue:**
```
moby build -output iso-bios linuxkit.yml 
linuxkit push vcenter -url https://user:password@192.168.122.2/sdk -datastore datastore1 -path linuxkit linuxkit.iso
```

**Describe the results you received:**

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x561fc388213c]

goroutine 1 [running]:
main.vCenterConnect(0x561fc4c10bc0, 0xc420017280, 0xc420190b90, 0xc420190ba0, 0xc420190bb0, 0x0, 0xc420190bc0, 0xc420190be0, 0xc420190bd0, 0x0, ...)
	/go/src/github.com/linuxkit/linuxkit/run_vcenter.go:204 +0x5ec
main.pushVCenter(0xc420010170, 0x7, 0x7)
	/go/src/github.com/linuxkit/linuxkit/push_vcenter.go:63 +0x585
main.push(0xc420010160, 0x8, 0x8)
	/go/src/github.com/linuxkit/linuxkit/push.go:45 +0x4a9
main.main()
	/go/src/github.com/linuxkit/linuxkit/main.go:86 +0x417
```

**Describe the results you expected:**
Pushing the iso to vcenter

**Additional information you deem important (e.g. issue happens only occasionally):**

The network is removed from the push command [here](https://github.com/linuxkit/linuxkit/commit/846e814a7fa464c017c1baaafadaf2d891b939ea#diff-953bd18379d8c020ea390e47bf0912f1)

and in the vcenter [code](https://github.com/linuxkit/linuxkit/blob/master/src/cmd/linuxkit/run_vcenter.go#L204) the network is checked while it does not exist.

**Comment by justincormack** [comment]
cc @sbourlon @thebsdbox

**Comment by firecyberice** [comment]
Fixed by #2673
